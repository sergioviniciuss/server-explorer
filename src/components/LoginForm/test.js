import React from "react";
import { render, screen, fireEvent, act } from "@testing-library/react";
import { LoginForm } from ".";

describe("LoginForm", () => {
  it("renders component properly", () => {
    const { container } = render(<LoginForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should display username input", () => {
    render(<LoginForm />);
    expect(screen.getByPlaceholderText(/username/i)).toBeInTheDocument();
  });

  it("Should display password input", () => {
    render(<LoginForm />);
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();
  });

  it("Should display a button with login text", () => {
    render(<LoginForm />);
    expect(
      screen.getByRole("button", {
        name: /login/i,
      })
    ).toBeInTheDocument();
  });
});
