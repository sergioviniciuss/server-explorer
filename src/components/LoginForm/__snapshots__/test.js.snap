// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders component properly 1`] = `
<div
  class="wrapper"
>
  <form
    class="container"
  >
    <input
      autocomplete="new-password"
      class="textField"
      placeholder="username"
      type="text"
      value=""
    />
    <input
      autocomplete="new-password"
      class="textField"
      placeholder="password"
      type="password"
      value=""
    />
    <button
      class="cta"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
